qiskit/chemistry/algorithms/pes_samplers/potentials/morse_potential.py

```diff
@@ -0,0 +1,247 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+This module implements a 1D Morse potential.
+"""
+
+from typing import List
+import numpy as np
+from scipy.optimize import curve_fit
+
+from qiskit.chemistry.algorithms.pes_samplers.potentials.potential_base import PotentialBase
+from qiskit.chemistry.drivers import Molecule
+import qiskit.chemistry.constants as const
+
+
+class MorsePotential(PotentialBase):
+    """
+    Implements a 1D Morse potential.
+    Input units are Angstroms (distance between the two atoms),
+        and output units are Hartrees (molecular energy).
+    """
+    # Works in Angstroms and Hartrees
+
+    def __init__(self, molecule):
+        """
+        Initializes the potential to the zero-function.
+        fit() should be used afterwards to fit the potential to
+            computed molecular energies.
+
+        Args:
+            molecule: the underlying molecule.
+
+        Raises:
+            ValueError: Only implemented for diatomic molecules
+        """
+        # Initialize with zero-potential.
+        # Later - fit energy values (fit)
+        self.d_e = 0.0
+        self.m_shift = 0.0
+        self.alpha = 0.0
+        self.r_0 = 0.0
+        self._m_a = molecule.masses[0]
+        self._m_b = molecule.masses[1]
+
+    @staticmethod
+    def fit_function(x: float,
+                     d_e: float,
+                     alpha: float,
+                     r_0: float,
+                     m_shift: float) -> float:
+        """Functional form of the potential.
+
+        Args:
+            x: x parameter of morse potential
+            d_e: d_e parameter of morse potential
+            alpha: alpha parameter of morse potential
+            r_0: r_0 parameter of morse potential
+            m_shift: m_shift parameter of morse potential
+
+        Returns:
+            potential functional form
+        """
+        # d_e (Hartree), alpha (1/Angstrom), r_0 (Angstrom)
+        return d_e * (1 - np.exp(-alpha * (x - r_0))) ** 2 + m_shift
+
+    def eval(self, x: float) -> float:
+        """After fitting the data to the fit function, predict the energy at a point x.
+
+        Args:
+            x: value to evaluate surface in
+
+        Returns:
+            value of surface in point x
+        """
+        # Expects Angstroms returns Hartrees
+        return self.fit_function(x, self.d_e,
+                                 self.alpha, self.r_0, self.m_shift)
+
+    def update_molecule(self, molecule: Molecule) -> None:
```

Comment:
The name seems like more update_masses would be the name since that is all that is taken from the molecule and stored. Or maybe update_from_molecule if its felt that one would want to be more future proof and other items might be pulled from it in future.

Reply:
super class is invoked now.